record-and-playback/core/lib/recordandplayback/generators/audio.rb

```diff
@@ -73,5 +73,70 @@ def self.create_audio_edl(events, archive_dir)
       return audio_edl
     end
 
+    def self.create_deskshare_audio_edl(events, deskshare_dir)
+      audio_edl = []
+
+      initial_timestamp = BigBlueButton::Events.first_event_timestamp(events)
+      final_timestamp = BigBlueButton::Events.last_event_timestamp(events)
+      filename = ""
+
+      # Initially start with silence
+      audio_edl << {
+        :timestamp => 0,
+        :audio => nil
+      }
+
+      events.xpath('/recording/event[@module="Deskshare" or (@module="bbb-webrtc-sfu" and (@eventname="StartWebRTCDesktopShareEvent" or @eventname="StopWebRTCDesktopShareEvent"))]').each do |event|

```

Comment:
I don't think the "Deskshare" module (I think that's the old java screensharing?) can ever have audio, so this should only look at the WebRTC events.